From d4081eee3d679283a1c5187574d16b34355898ef Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Fri, 29 Sep 2023 11:46:05 +0200
Subject: [PATCH 0124/1695] pm: add macro for PM_HINT_STATE access

Add macro for PM_HINT_STATE access when PM_HINT_PLATFORM_STATE_MASK
is used by platform.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 2cd2217bc5f00d418feae05d83341b183dfe318c pm: add macro for PM_HINT_STATE access]
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: If9400986d87d9aaa8519c19fa2d9c2db4df6450c
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/330878
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 core/include/kernel/pm.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/core/include/kernel/pm.h b/core/include/kernel/pm.h
index 2badce965..8f9a01541 100644
--- a/core/include/kernel/pm.h
+++ b/core/include/kernel/pm.h
@@ -30,6 +30,12 @@
 #define PM_HINT_PLATFORM_STATE_MASK	GENMASK_32(31, 16)
 #define PM_HINT_PLATFORM_STATE_SHIFT	U(16)
 
+#define PM_HINT_STATE(x)		((x) & ~PM_HINT_PLATFORM_STATE_MASK)
+#define PM_HINT_PLATFORM_STATE(x) \
+	(((x) & PM_HINT_PLATFORM_STATE_MASK) >> PM_HINT_PLATFORM_STATE_SHIFT)
+
+#define PM_HINT_IS_STATE(x, name) (!!((x) & PM_HINT_ ## name ## _STATE))
+
 /*
  * PM_OP_SUSPEND: platform is suspending to a target low power state
  * PM_OP_RESUME: platform is resuming from low power state
-- 
2.47.2

